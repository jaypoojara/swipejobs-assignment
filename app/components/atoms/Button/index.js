import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Button } from 'react-native-paper';
import colors from '../../../themes/colors';
import T from '../T';

const index = ({
  mode = 'contained',
  color = colors.black,
  textColor = colors.white,
  messageId,
  onPress
}) => {
  return (
    <Button
      mode={mode}
      color={color}
      uppercase={false}
      style={styles.buttonContainer}
      contentStyle={styles.buttonHeight}
      onPress={onPress}
    >
      <T id={messageId} style={{ color: textColor }} />
    </Button>
  );
};

export default index;

const styles = StyleSheet.create({
  buttonContainer: {
    width: '45%'
  },
  buttonHeight: {
    height: 50
  }
});
