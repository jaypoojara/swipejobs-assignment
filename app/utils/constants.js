export const NOTIFICATION_TYPE = {
  DEFAULT: 0,
  SUCCESS: 1,
  ERROR: 2,
  WARNING: 3,
  INFO: 4
};

export const WORKER_ID = '7f90df6e-b832-44e2-b624-3143d428001f';
