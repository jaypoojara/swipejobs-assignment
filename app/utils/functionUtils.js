import moment from 'moment';
import momentTz from 'moment-timezone';

export const getDuration = (startDate, endDate) => {
  const days = moment(startDate)
    .local()
    .format('MMM DD, ddd HH:mm A');
  const endDateTemp = moment(endDate)
    .local()
    .format('HH:mm A');
  const timeZone = momentTz.tz.guess();
  const timeZoneAbr = momentTz.tz(timeZone).zoneAbbr();
  return `${days} - ${endDateTemp} ${timeZoneAbr}`;
};
