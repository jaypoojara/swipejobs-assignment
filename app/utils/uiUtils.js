import { showMessage } from 'react-native-flash-message';
import { NOTIFICATION_TYPE } from './constants';
export function showMessageToUser({ title, message, notificationType }) {
  let messageType;
  switch (notificationType) {
    case NOTIFICATION_TYPE.DEFAULT:
      messageType = 'default';
      break;
    case NOTIFICATION_TYPE.INFO:
      messageType = 'info';
      break;
    case NOTIFICATION_TYPE.SUCCESS:
      messageType = 'success';
      break;
    case NOTIFICATION_TYPE.ERROR:
      messageType = 'danger';
      break;
    case NOTIFICATION_TYPE.WARNING:
      messageType = 'warning';
      break;
    default:
      messageType = 'none';
  }
  setTimeout(() => {
    showMessage({
      message: title,
      description: message,
      type: messageType
    });
  }, 100);
}
