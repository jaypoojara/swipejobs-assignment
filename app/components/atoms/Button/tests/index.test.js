import React from 'react';
import { renderProvider } from '../../../../utils/testUtils';
import Button from '../index';

describe('<Button /> Component tests', () => {
  it('should render and match the button snapshot', () => {
    const baseElement = renderProvider(<Button />);
    expect(baseElement).toMatchSnapshot();
  });
});
