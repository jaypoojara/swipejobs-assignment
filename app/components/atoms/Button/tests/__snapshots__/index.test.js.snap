// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Component tests should render and match the button snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000000",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "elevation": 2,
      "minWidth": 64,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.75,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 1.5,
      "width": "45%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "height": 50,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              false,
              Object {
                "color": "#ffffff",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
            }
          }
          testID="t"
        >
          
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
